src/cellar_uniswap_wrapper.rs

```diff
@@ -32,9 +26,16 @@ impl<T: 'static + Middleware> UniswapV3CellarState<T> {
     }
 
     // Rebalance portfolio with cellar tick info
-    pub async fn rebalance(&mut self, cellar_tick_info: Vec<UniswapV3CellarTickInfo>) -> Result<(), Error> {
-        let mut ticks: Vec<(U256, i32, i32, u32)> =
-            cellar_tick_info.into_iter().map(|x| x.to_tuple()).collect();
+    pub async fn rebalance(&mut self, cellar_tick_info: Vec<CellarTickInfo>) -> Result<(), Error> {
+        let mut ticks: Vec<CellarTickInfo> = cellar_tick_info
```

Comment:
After removing the wrapper, the argument is already of the type CellarTickInfo, so you could just make a mutable clone instead.